src/builder/webpack/config.browser.js

```diff
@@ -1,21 +1,36 @@
 // @flow
 
 import { join } from "path"
-
+import findCacheDir from "find-cache-dir"
 import commonWebpackConfig from "./config.common.js"
 import { offlinePlugin, offlineEntry } from "../../_utils/offline/webpack.js"
+import HardSourceWebpackPlugin from "hard-source-webpack-plugin"
 
 const chunkNameBrowser = "phenomic.browser"
 
 export default (config: PhenomicConfig): WebpackConfig => {
+  // Use different cache folder for hard-source-webpack-plugin on each env
+  const env = (config.production) ? "prod" : "dev"
+  const cacheDir = findCacheDir({ name: "phenomic-hard-source-wp" + env })
+
   const webpackConfig = commonWebpackConfig(config)
 
   return {
     ...webpackConfig,
-
+    recordsPath: join(cacheDir, "records.json"),
     plugins: [
       ...webpackConfig.plugins,
       ...offlinePlugin(config),
+      new HardSourceWebpackPlugin({
+        cacheDirectory: join(cacheDir, "cacheDirectory"),
+        environmentPaths: {
+          root: config.cwd,
+          files: [
+            "package.json",
+            "webpack.config.babel.js",
+          ],
+        },
+      }),
```

Comment:
can you move that in a another file like I did for offline option? Also, that would be nice to enable this behind a flag, so people can disable it if there is any issue with their webpack config.
Also if could be nice to have this for static build too, so should more be in common config.